proxy/tls.go

```diff
@@ -124,35 +125,131 @@ func newTLSListenerConfig(conf *config.Config) (*tls.Config, error) {
 
 	cfg.VerifyPeerCertificate = func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
 		if conf.Proxy.TLS.ClientCert.ValidateSubject {
-			expected := fmt.Sprintf("s:/CN=%s/C=%v/S=%v/L=%v/O=%v/OU=%v",
-				conf.Proxy.TLS.ClientCert.Subject.CommonName,
-				conf.Proxy.TLS.ClientCert.Subject.Country,
-				conf.Proxy.TLS.ClientCert.Subject.Province,
-				conf.Proxy.TLS.ClientCert.Subject.Locality,
-				conf.Proxy.TLS.ClientCert.Subject.Organization,
-				conf.Proxy.TLS.ClientCert.Subject.OrganizationalUnit)
+
+			expectedFields := map[string]string{}
```

Comment:
expectedFields could be calculated once outside VerifyPeerCertificate function invocation

Reply:
I need a couple of sentences of guidance. The only way to precalculate this would be to do this somewhere on a config. But this would require a config private visibility property. Is that how you would go about it?

Looking at existing `config.Config`, I can't see any examples of such usage.

Is this a problem when this is executed only when creating a new TLS listener?

Reply:
I thought something like this (this is only an example snippet)

```
func newTLSListenerConfig(conf *config.Config) (*tls.Config, error) {
...
cfg.VerifyPeerCertificate = verifyPeerCertificate(conf)
...
}

func verifyPeerCertificate(conf *config.Config) func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error{
	// this is constant and it will not be changed
	expectedFields := getPeerCertificateExpectedFields(conf)

	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
		for _, chain := range verifiedChains {
				for _, cert := range chain {
					...
					for k, v := range expectedFields {
					...
					}
				}
		}
		...
	}
}

func getPeerCertificateExpectedFields(conf *config.Config) map[string]string {
	....
}

```

Reply:
Done https://github.com/grepplabs/kafka-proxy/pull/65/files#diff-2ee690fbbffb319d2291746fea4d8eeeR221.